docs/userguide/glossary.rst

```diff
@@ -0,0 +1,218 @@
+
+.. _elasticityglossary:
+
+**Elasticity:**
+-----------------
+
+Elasticity means being able to stretch or shrink. In Parsl, it means that you can easily add or remove blocks of computers as needed. This is useful if your workload changes – you can use more computers when you have a lot of work to do and fewer computers when you don't.
+
+.. _executionproviderglossary:
+
+**Execution Provider:**
+--------------------------
+
+An execution provider is like a bridge between Parsl and the computers you want to use. It's the part that knows how to talk to different types of computers, like your laptop, a cluster, or a cloud service.
+
+.. _executorglossary:
+
+**Executor:**
+----------------
+
+An executor is like a manager for your apps. It's the part that decides which app should run on which computer and when. It's like a traffic controller, directing the flow of apps to make sure they all get where they need to go.
+
+.. _futureglossary:
+
+**Future:**
+-------------
+
+A future is a placeholder for the result of a task that hasn't finished yet. Both AppFuture and DataFuture are types of Futures. You can use the ``.result()`` method to get the actual result when it's ready.
+
+.. _jobglossary:
+
+**Job:**
+---------
+
+A job in Parsl refers to a unit of work that is submitted to an execution environment (like a cluster or cloud) for processing. It's like a task that needs to be done, such as running a script or processing data, and it can consist of one or more tasks that are executed on a compute resource.
+
+.. _launcherglossary:
+
+**Launcher:**
+----------------
+
+A launcher in Parsl is responsible for placing the workers onto each computer, preparing them to run the apps. It’s like a bus driver who brings the players to the stadium, ensuring they are ready to start, but not directly involved in telling them what to do once they arrive.
+
+.. _managerglossary:
+
+**Manager:**
+--------------
+
+A manager in Parsl is responsible for overseeing the execution of tasks on specific compute resources. It's like a supervisor who ensures that all workers (or workers within a block) are carrying out their tasks correctly and efficiently.
+
+.. _memoizationglossary:
+
+**Memoization:**
+-------------------
+
+Memoization is like remembering something so you don't have to do it again. In Parsl, it means that if you run an app with the same inputs multiple times, Parsl will remember the result from the first time and give it to you again instead of running the app again. This can save a lot of time.
+
+.. _mpiappglossary:    
+
+**MPI App:**
+---------------
+
+An MPI app is a special kind of app that uses a technology called Message Passing Interface (MPI) to communicate between different computers. It's like a walkie-talkie that lets different apps talk to each other.
```

Comment:
Is "computers" the right word here?

Reply:
The second sentence implies communication between apps, while MPI is really communication within an app.